### DISCUSSION TITLE
AppService/Vibration works haphazardly

### DISCUSSION THREAD

**Discussion by vxsw** [zepp-health/zeppos-forum#288] [Q&A]
hi all,

I'll tried to make a little watchdog for bt connection in background and it works at least strange.
through few minutes after installing it works fine, but after few hours, it may works or may not, as it pleases. no logic or dependencies. also it works only while display is off.
at first time, I'd thought what it's a listener issue, but injecting of timer-checker patch hand no affects.
so is I do something wrong or I do not understand how it must works?

`
import { Vibrator } from "@zos/sensor";
import { addListener, connectStatus  } from '@zos/ble'

const vibrator = new Vibrator()

let isConnected = connectStatus()

const callback_btListener = (cState) => {
  isConnected = cState
  doVibro()
}
addListener(callback_btListener)

function doVibro(){
  if (isConnected){
    vibrator.stop()
    vibrator.setMode(31)
    vibrator.start()
  } else {
    vibrator.stop()
    vibrator.setMode(23)
    vibrator.start()
  }
}

setInterval(() => {
  if (isConnected != connectStatus()){
    isConnected = connectStatus()
    doVibro()
  }
}, 1000)

AppService({
  onEvent(e) {
  },
  onInit(e) {
  },
  onDestroy() {
  },
});
`

**Comment by MyoungXUE** [comment]
Hello, I have some information to share with you

1. In App Service, the timer-related interface cannot be used. You can check https://docs.zepp.com/docs/guides/framework/device/app-service/#system-api-capabilities-and-limitations

![image](https://github.com/zepp-health/zeppos-forum/assets/107526244/958c1e82-2f61-4b2a-ba9b-9981438620e3)

2. As an alternative to timers, onMinuteChange of Time Sensor can be used to implement a timer that runs once a minute

**Comment by vxsw** [comment]
oh. thanks, I'll try onminute, but anyway, timer used as patch only, main function is a listener and vibration.
